src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/Custom/EES1598RemoveChartLabelsField.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using GovUk.Education.ExploreEducationStatistics.Common.Model.Chart;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Microsoft.EntityFrameworkCore;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Migrations.Custom
+{
+    public class EES1598RemoveChartLabelsField : ICustomMigration
+    {
+        private readonly ContentDbContext _context;
+        private readonly bool _isTest;
+
+        public EES1598RemoveChartLabelsField(ContentDbContext context, bool isTest = false)
+        {
+            _context = context;
+            _isTest = isTest;
+        }
+
+        public void Apply()
+        {
+            var dataBlocks = GetQuery();
+
+            foreach (var dataBlock in dataBlocks)
+            {
+                var chart = dataBlock.Charts.ElementAtOrDefault(0);
+
+                // No labels, this is data block is
+                // already in the correct format.
+                if (chart == null || chart.Type == ChartType.Infographic || chart.Labels == null)
+                {
+                    continue;
+                }
+
+                foreach (var (key, config) in chart.Labels)
+                {
+                    // The key is a junk label so we can ignore it.
+                    // We use a length of 37 as this is a single
+                    // indicator guid and ____ concatenated
+                    // (the bare minimum to be considered a data set).
+                    if (key.Length < 37 || !key.EndsWith("_____"))
+                    {
+                        continue;
+                    }
+
+                    var labelDataSet = ParseLabelDataSet(key, config);
+                    MigrateLabelDataSet(chart, labelDataSet);
+                }
+
+                chart.Labels = null;
```

Comment:
I guess in the case that we hit the code block:

```
            // This chart has no major axis for some reason so we can ignore it
            if (chart.Axes?.GetValueOrDefault("major") == null)
            {
                return;
            }
```

in `MigrateLabelDataSet(chart, labelDataSet)`, then no actual label migration occurs.  We're then nulling out the Labels on this line despite not having migrated anything.

Is this OK to do?

Reply:
I suppose what I'm actually asking is what's the situation where we have Labels but no major axis, and in that particular situation are we risking losing any important Label information by nulling them out at this point?  I'm guessing that the answer is "no, it'll all just be junk labels", but just wanted to check.

Reply:
So in a situation where the major axis is null, the chart will be unable to render anyway so it will just be a broken chart no matter what. The best course of action will just be to recreate the chart.

On prod we'll be migrating about 20-30 charts from releases that should have already been released by now, so it's unlikely that they would have been published without a major axis.

Reply:
Thanks!